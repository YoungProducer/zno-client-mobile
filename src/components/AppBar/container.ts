/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 2 March 2020
 *
 * Declare main types and interfaces for AppBar component.
 * Create function which connect actions
 * and/or variables from the redux-store to the component.
 */

/** External imports */
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStyles, Theme, withStyles, WithStyles } from '@material-ui/core/styles';
import { green } from '@material-ui/core/colors';

/** Application's imports */

/** Define Material ui styles */
const styles = (theme: Theme) =>
    createStyles({
        root: {
            background: green[700],
            width: '100%',
            height: 50,
        },
        container: {
            display: 'flex',
            justifyContent: 'center',
            alignItems: 'center',
            height: 'inherit',
        },
        navLink: {
            color: '#fff',
            fontSize: '1.2rem',
            marginRight: theme.spacing(3),
            '&:last-child': {
                marginRight: 0,
            },
        },
    });

/** Props which component get from the parent */
interface IOwnProps extends WithStyles<typeof styles>{}

/** Declare type which describe all props pushed to the component */
export type TAppBarProps =
    IOwnProps;

/** Export function to configure component via HOC */
export default compose(
    withStyles(styles, { withTheme: true }),
);
