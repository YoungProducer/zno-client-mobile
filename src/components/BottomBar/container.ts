/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 28 March 2020
 *
 * Container for BottomBar component.
 */

/** External imports */
import { connect } from 'react-redux';

/** Application's imports */
import { selectIsLoggedIn } from 'store/selectors/auth';
import { fetchLogoutAction } from 'store/actionCreators/auth';
import { RootState, setTestSuiteFinishedAction } from 'store/slices';

interface IStateProps {
    loggedIn: boolean;
}

interface IDispatchProps {
    logout: () => void;
    setTestSuiteFinished: (value: boolean) => void;
}

export type TBottomBarProps =
    IStateProps
    & IDispatchProps;

const mapStateToProps = (state: RootState): IStateProps => ({
    loggedIn: selectIsLoggedIn(state),
});

const mapDispatchToProps = (dispatch: any): IDispatchProps => ({
    logout: () => dispatch(fetchLogoutAction()),
    setTestSuiteFinished: (finished: boolean) =>
        dispatch(setTestSuiteFinishedAction(finished)),
});

export default connect<IStateProps, IDispatchProps>(
    mapStateToProps,
    mapDispatchToProps,
);
