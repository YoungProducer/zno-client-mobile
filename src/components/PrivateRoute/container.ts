/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 1 March 2020
 *
 * Define main types and interface related to PrivateRoute component.
 * Create function which connect variables from the redux store.
 */

/**External imports */
import { ReactNode } from 'react';
import { connect } from 'react-redux';
import { RouteProps } from 'react-router-dom';

// Applicaiton's imports
import { selectIsLoggedIn } from 'store/selectors/auth/signin';
import { RootState } from 'store/slices';

/**
 * Props which component get from the parent component.
 */
interface IOwnProps extends RouteProps {
    children: ReactNode;
}

/**
 * Props which component get from the redux store.
 */
interface IStateProps {
    isLoggedIn: boolean;
}

/**
 * Declare type which describe all props pushed to the component.
 */
export type TPrivateRoute =
    IOwnProps
    & IStateProps;

/**
 * Define function mapStateToProps.
 * Function select variables from the redux store.
 */
const mapStateToProps = (state: RootState): IStateProps => ({
    isLoggedIn: selectIsLoggedIn(state),
});

/**
 * Export function which connect actions
 * or/and variables from the redux store to component.
 */
export default connect<IStateProps, {}, IOwnProps>(
    mapStateToProps,
);
