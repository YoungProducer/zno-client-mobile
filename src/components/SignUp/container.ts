/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 27 February 2020
 *
 * Create function which connect actions
 * and/or variables from the redux-store to the component.
 */

// External imports
import { connect } from 'react-redux';

// Application's imports
import { fetchSignUpAction } from 'store/actionCreators/auth/signup';
import {
    setSignUpErrorFieldsToDefaultAction,
    setSignUpFieldsMessagesToDefaultAction,
    ISignUpErrorFields,
    ISignUpFieldsMessages,
} from 'store/slices/auth';
import {
    selectSignUpLoading,
    selectSignUpErrorFields,
    selectSignUpFieldsMessages,
} from 'store/selectors/auth';
import { IFetchSignUpActionCredentials } from 'utils/verify-credentials';
import { RootState } from 'store/slices';

/**
 * Props which component get from the parent.
 */
interface IOwnProps {}

/**
 * Props which component get from the redux-store.
 */
interface IStateProps {
    loading: boolean;
    errorFields: ISignUpErrorFields;
    fieldsMessages: ISignUpFieldsMessages;
}

/**
 * Props(actions) which component can dispatch.
 */
interface IDispatchProps {
    fetchSignUp: (credentials: IFetchSignUpActionCredentials) => void;
    setSignUpErrorFieldsToDefault: () => void;
    setSignUpFieldsMessagesToDefault: () => void;
}

/**
 * Export type which describe all props pushed to the component.
 */
export type TSignUpProps = IOwnProps & IStateProps & IDispatchProps;

/**
 * Connect variables from the store to component.
 */
const mapStateToProps = (state: RootState): IStateProps => ({
    loading: selectSignUpLoading(state),
    errorFields: selectSignUpErrorFields(state),
    fieldsMessages: selectSignUpFieldsMessages(state),
});

/**
 * Connect actions which component must dispatch.
 */
const mapDispatchToProps = (dispatch: any): IDispatchProps => ({
    fetchSignUp: (credentials: IFetchSignUpActionCredentials) =>
        dispatch(fetchSignUpAction(credentials)),

    setSignUpErrorFieldsToDefault: () =>
        dispatch(setSignUpErrorFieldsToDefaultAction()),

    setSignUpFieldsMessagesToDefault: () =>
        dispatch(setSignUpFieldsMessagesToDefaultAction()),
});

// Export function which connect acions and/or variables from the redux-store to the component.
export default connect<IStateProps, IDispatchProps, IOwnProps>(
    mapStateToProps,
    mapDispatchToProps,
);
