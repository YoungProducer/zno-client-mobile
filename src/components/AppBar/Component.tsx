/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 2 March 2020
 *
 * Header component which contain main routes and another information or actions.
 */

/** External imports */
import React from 'react';
import CssBaseLine from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';
import AppBar from '@material-ui/core/AppBar';

/** Application's imports */
import { TAppBarProps } from './container';
import NavigationLink from 'components/NavigationLink';

/** Create component */
const Component = (props: TAppBarProps) => {
    const { classes } = props;

    return (
        <React.Fragment>
            <CssBaseLine />
            <AppBar
                position='static'
                className={classes.root}
            >
                <Container
                    maxWidth='lg'
                    className={classes.container}
                >
                    <NavigationLink
                        navLink={{
                            to: '/',
                        }}
                        className={classes.navLink}
                        data-testid='home-navlink'
                    >
                        Головна
                    </NavigationLink>
                    <NavigationLink
                        navLink={{
                            to: '/subject-selection',
                        }}
                        className={classes.navLink}
                        data-testid='tests-navlink'
                    >
                        Тести
                    </NavigationLink>
                </Container>
            </AppBar>
        </React.Fragment>
    );
};

export default Component;
