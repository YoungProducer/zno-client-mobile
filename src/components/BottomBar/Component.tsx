/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 28 March 2020
 *
 * Component which display main navigation of app.
 */

/** External imports */
import React, { useState, useMemo } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import classNames from 'classnames';
import Collapse from '@material-ui/core/Collapse';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import ExpandLessIcon from '@material-ui/icons/ExpandLess';
import HomeIcon from '@material-ui/icons/Home';
import MoreHorizIcon from '@material-ui/icons/MoreHoriz';
import PersonIcon from '@material-ui/icons/Person';
import { makeStyles, createStyles, Theme } from '@material-ui/core';

/** Application's imports */
import { TBottomBarProps } from './container';

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            position: 'absolute',
            minHeight: 70,
            borderRadius: 9,
            background: '#fff',
            filter: 'drop-shadow(0px 3px 3.5px rgba(0,0,0,0.16))',
            bottom: theme.spacing(1),
            left: theme.spacing(1),
            right: theme.spacing(1),
            transform: `translateY(0)`,
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        rootHidden: {
            transform: `translateY(60%)`,
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        innerContainer: {
            width: '100%',
            height: '100%',
        },
        actionsContainer: {
            width: '100%',
            height: '100%',
            minHeight: 70,
            display: 'flex',
            justifyContent: 'space-around',
            alignItems: 'center',
        },
        expandIconButton: {
            zIndex: 10,
            position: 'absolute',
            left: '50%',
            transform: `translate(-50%, -60%)`,
            borderRadius: '50%',
            background: '#fff',
            filter: 'drop-shadow(0px 3px 3.5px rgba(0,0,0,0.05))',
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
            '&:hover': {
                background: '#fff',
            },
        },
        expandIconButtonHidden: {
            transform: `translate(-50%, -35%)`,
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        expandIcon: {
            transform: `rotateX(180deg)`,
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        expandIconHidden: {
            transform: `rotateX(0deg)`,
            transition: theme.transitions.create(['transform'], {
                duration: 200,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        actionIconButton: {
            background: '#fff',
            boxShadow: `0px 3px 3.5px rgba(0,0,0,0)`,
            transition: theme.transitions.create(['box-shadow'], {
                duration: 300,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        actionIconButtonActive: {
            boxShadow: `0px 3px 3.5px rgba(0,0,0,0.16)`,
            transition: theme.transitions.create(['box-shadow'], {
                duration: 500,
                easing: theme.transitions.easing.easeInOut,
            }),
        },
        actionIcon: {
            color: '#69deac',
            width: 35,
            height: 35,
        },
        expandActions: {
            padding: theme.spacing(1),
            display: 'flex',
            justifyContent: 'space-between',
        },
        button: {
            color: '#fff',
            background: '#656cff',
            borderRadius: 4,
            width: '45%',
        },
    }));

const Component = ({ loggedIn, logout }: TBottomBarProps) => {
    const classes = useStyles({});

    const history = useHistory();
    const location = useLocation();

    const [hidden, setHidden] = useState<boolean>(false);
    const [expand, setExpand] = useState<boolean>(false);

    const toggleHidden = () => {
        if (expand) {
            setExpand(false);
            window.setTimeout(() => {
                setHidden(!hidden);
            }, 300);
        } else {
            setHidden(!hidden);
        }
    };
    const toggleExpand = () => setExpand(!expand);

    const handleSignIn = () => history.push('/auth/signin');
    const handleSignUp = () => history.push('/auth/signup');
    const handleHome = () => history.push('/');
    const handleProfile = () => history.push('/user/profile');

    const activeIcons = useMemo(() => {
        const { pathname } = location;

        return {
            home: pathname === '/' || pathname === '/subject-selection',
            profile: pathname === '/user/profile',
        };
    }, [location]);

    return (
        <div
            className={classNames(classes.root, {
                [classes.rootHidden]: hidden,
            })}
        >
            <IconButton
                className={classNames(classes.expandIconButton, {
                    [classes.expandIconButtonHidden]: hidden,
                })}
                onClick={toggleHidden}
            >
                <ExpandLessIcon className={classNames(classes.expandIcon, {
                    [classes.expandIconHidden]: hidden,
                })}/>
            </IconButton>
            <div className={classes.innerContainer}>
                <div className={classes.actionsContainer}>
                    <IconButton
                        disabled={hidden}
                        className={classNames(classes.actionIconButton, {
                            [classes.actionIconButtonActive]: activeIcons.profile,
                        })}
                        onClick={handleProfile}
                    >
                        <PersonIcon className={classes.actionIcon}/>
                    </IconButton>
                    <IconButton
                        disabled={hidden}
                        className={classNames(classes.actionIconButton, {
                            [classes.actionIconButtonActive]: activeIcons.home,
                        })}
                        onClick={handleHome}
                    >
                        <HomeIcon className={classes.actionIcon}/>
                    </IconButton>
                    <IconButton
                        disabled={hidden}
                        className={classNames(classes.actionIconButton, {
                            [classes.actionIconButtonActive]: expand,
                        })}
                        onClick={toggleExpand}
                    >
                        <MoreHorizIcon className={classes.actionIcon}/>
                    </IconButton>
                </div>
                <Collapse in={expand} unmountOnExit>
                    <div className={classes.expandActions}>
                        { loggedIn && (
                            <Button
                                className={classes.button}
                                variant='contained'
                                disableElevation
                                onClick={logout}
                            >
                                Вийти
                            </Button>
                        )}
                        { !loggedIn && (
                            <>
                                <Button
                                    className={classes.button}
                                    variant='contained'
                                    disableElevation
                                    onClick={handleSignIn}
                                >
                                    Увійти
                                </Button>
                                <Button
                                    className={classes.button}
                                    variant='contained'
                                    disableElevation
                                    onClick={handleSignUp}
                                >
                                    Зареєструватися
                                </Button>
                            </>
                        )}
                    </div>
                </Collapse>
            </div>
        </div>
    );
};

export default Component;
