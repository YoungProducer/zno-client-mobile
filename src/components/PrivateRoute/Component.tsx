/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 1 March 2020
 *
 * Permission route which automaticaly redirect from protected pages if user is not logged in to SignIn page.
 */

/** External imports */
import React from 'react';
import { Redirect, Route } from 'react-router-dom';

/** Application's imports */
import { TPrivateRoute } from './container';

/** Create component */
const Component = ({ isLoggedIn, children, ...rest }: TPrivateRoute) => (
    <Route
        {...rest}
        render={({ location }) => {
            const pathname = location
                ? `/auth/signin?next=${location.pathname}${location.search}`
                : '/auth/signin';

            return isLoggedIn
                ? children
                : <Redirect
                    to={{
                        pathname: '/auth/signin',
                        state: { from: location },
                    }}
                />;
        }
        }
    />
);

export default Component;
