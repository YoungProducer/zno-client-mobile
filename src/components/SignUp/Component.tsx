/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 25 February 2020
 *
 * Component which allows create user's account.
 */

// External imports
import React, { useState, useCallback, useEffect, memo, useMemo } from 'react';
import Container from '@material-ui/core/Container';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import EmailIcon from '@material-ui/icons/Email';
import LockIcon from '@material-ui/icons/Lock';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

// Application's imports
import { TSignUpProps } from './container';
import NavigationLink from 'components/NavigationLink';
import Input from 'components/custom/Input';
import Logo from 'img/logo';

// Define classes as hook
const useStyles = makeStyles((theme: Theme) => createStyles({
    backdrop: {
        background: '#fff',
        width: '100%',
        height: '100vh',
    },
    container: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
        height: '100%',
    },
    root: {
        width: 'min-content',
        borderRadius: 4,
        filter: `drop-shadow(2.828px 2.828px 3.5px rgba(227,207,207,0.9))`,
        backgroundColor: '#ffffff',
        textAlign: 'center',
    },
    innerContainer: {
        borderRadius: 4,
        backgroundColor: '#f4f5f7',
        padding: theme.spacing(4),
    },
    logoContainer: {
        paddingTop: theme.spacing(1),
        paddingBottom: theme.spacing(1),
    },
    textField: {
        width: 230,
    },
    fields: {
        paddingBottom: theme.spacing(3),
    },
    actions: {
        display: 'flex',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    block: {
        width: 'max-content',
    },
    button: {
        filter: `drop-shadow(0px 3px 2.5px rgba(0,0,0,0.16))`,
    },
    link: {
        color: theme.palette.primary.main,
    },
    icon: {
        color: '#555',
    },
}));

/**
 * Create hook to make component stateless.
 * Compute all values from text fields.
 * Returns fields data and button onClick property.
 */
const useSignUpElements = (props: TSignUpProps) => {
    const {
        errorFields,
        fieldsMessages,
        fetchSignUp,
        setSignUpErrorFieldsToDefault,
        setSignUpFieldsMessagesToDefault,
    } = props;

    const clearFields = () => {
        setSignUpErrorFieldsToDefault();
        setSignUpFieldsMessagesToDefault();
    };

    /** User email */
    const [email, setEmail] = useState<string>('');
    const handleChangeEmail =
        useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
            setEmail(event.target.value);
            if (errorFields.email) {
                clearFields();
            }
        }, [errorFields.email]);

    /** User password */
    const [password, setPassword] = useState<string>('');
    const handleChangePassword =
        useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
            setPassword(event.target.value);
            if (errorFields.password) {
                clearFields();
            }
        }, [errorFields.password]);

    /** Confirmation password, should be the same to the password */
    const [confPassword, setConfPassword] = useState<string>('');
    const handleChangeConfPassword =
        useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
            setConfPassword(event.target.value);
            if (errorFields.confPassword) {
                clearFields();
            }
        }, [errorFields.confPassword]);

    /** SignUp button handle */
    const handleFetchSignUp = useCallback(() => fetchSignUp({
        email,
        password,
        confPassword,
    }), [email, password, confPassword]);

    return {
        emailField: {
            onChange: handleChangeEmail,
            value: email,
            error: errorFields.email,
            helperText: fieldsMessages.email,
        },
        passwordField: {
            onChange: handleChangePassword,
            value: password,
            error: errorFields.password,
            helperText: fieldsMessages.password,
        },
        confPasswordField: {
            onChange: handleChangeConfPassword,
            value: confPassword,
            error: errorFields.confPassword,
            helperText: fieldsMessages.confPassword,
        },
        signUpButton: {
            onClick: handleFetchSignUp,
        },
    };
};

const Component = (props: TSignUpProps) => {
    /** Create object with classes to use them */
    const classes = useStyles({});

    /** Get fields data from hook */
    const {
        emailField,
        passwordField,
        confPasswordField,
        signUpButton,
    } = useSignUpElements(props);

    return (
        <div className='auth-background'>
            <Container maxWidth='lg' className={classes.container}>

                <Paper
                    className={classes.root}
                    elevation={0}
                    square
                >
                    <div className={classes.logoContainer}>
                        <Logo />
                    </div>
                    <div className={classes.innerContainer}>
                        <Grid
                            container
                            direction='column'
                            alignItems='center'
                            item
                            spacing={3}
                            className={classes.block}
                        >
                            <Grid item>
                                <Input
                                    color='primary'
                                    type='email'
                                    placeholder='Емеїл'
                                    data-testid='signup-email-input'
                                    startAdornment={<EmailIcon className={classes.icon}/>}
                                    { ...emailField }
                                />
                            </Grid>
                            <Grid item>
                                <Input
                                    color='primary'
                                    type='password'
                                    placeholder='Пароль'
                                    data-testid='signup-password-input'
                                    startAdornment={<LockIcon className={classes.icon}/>}
                                    { ...passwordField }
                                />
                            </Grid>
                            <Grid item>
                                <Input
                                    color='primary'
                                    type='password'
                                    placeholder='Пароль підтвердження'
                                    data-testid='signup-conf-password-input'
                                    startAdornment={<LockIcon className={classes.icon}/>}
                                    { ...confPasswordField }
                                />
                            </Grid>
                            <Grid item>
                                <Button
                                    color='primary'
                                    variant='contained'
                                    disableElevation
                                    className={classes.button}
                                    data-testid='signup-button'
                                    { ...signUpButton }
                                >
                                    Зареєструватися
                                </Button>
                            </Grid>
                            <Grid>
                                <Typography align='center'>
                                    Вже зареєстровані?
                                    <NavigationLink
                                        navLink={{
                                            to: '/auth/signin',
                                        }}
                                        className={classes.link}
                                    >
                                        Увійти
                                    </NavigationLink>
                                </Typography>
                            </Grid>
                        </Grid>
                    </div>
                </Paper>
            </Container>
        </div>
    );
};

export default Component;
