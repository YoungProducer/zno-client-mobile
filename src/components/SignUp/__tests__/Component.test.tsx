/**
 * Created by: Oleksandr Bezrukov
 * Creation date: 26 February 2020
 *
 * Create test suites for SignUp Component.
 * @jest-environment jsdom
 */

// External imports
import React from 'react';
import { shallow, mount } from 'enzyme';
import MockAdapter from 'axios-mock-adapter';

// Application's imports
import api from 'api';
import Component from '../Component';
import { TSignUpProps } from '../container';

describe('SignUp component', () => {
    // Mock props functions
    const fetchSignUp = jest.fn();
    const setSignUpErrorFieldsToDefault = jest.fn();
    const setSignUpFieldsMessagesToDefault = jest.fn();

    const requiredProps: TSignUpProps = {
        fetchSignUp,
        setSignUpErrorFieldsToDefault,
        setSignUpFieldsMessagesToDefault,
        errorFields: {
            email: false,
            password: false,
            confPassword: false,
        },
        fieldsMessages: {
            email: '',
            password: '',
            confPassword: '',
        },
        loading: false,
    };

    beforeEach(() => {
        (requiredProps.fetchSignUp as jest.Mock).mockReset();
        (requiredProps.setSignUpErrorFieldsToDefault as jest.Mock).mockReset();
        (requiredProps.setSignUpFieldsMessagesToDefault as jest.Mock).mockReset();
    });

    test('Is matches snapshot', () => {
        // Render shallow component
        const root = shallow(<Component {...requiredProps}/>);

        // Check is component matches snapshot
        expect(root).toMatchSnapshot();
    });

    test('Is fetchSignUp will be called with right args', () => {
        /** Mount component */
        const wrapper = shallow(<Component {...requiredProps}/>);

        /** Simulate onChange events */
        wrapper
            .find(`[data-testid='signup-email-input']`)
            .simulate('change', { target: { value: 'foo' } });
        wrapper
            .find(`[data-testid='signup-password-input']`)
            .simulate('change', { target: { value: 'bar' } });
        wrapper
            .find(`[data-testid='signup-conf-password-input']`)
            .simulate('change', { target: { value: 'abc' } });

        /** Simulate sign up button click */
        wrapper
            .find(`[data-testid='signup-button']`)
            .simulate('click');

        /** Assert fetchSignUp have been called with right args */
        expect(fetchSignUp).toBeCalledWith({
            email: 'foo',
            password: 'bar',
            confPassword: 'abc',
        });
    });

    test(`Is makes api call to signup endpoint when press button 'Реєстрація'`, () => {
        /** Mount component */
        const wrapper = shallow(<Component {...requiredProps}/>);

        /** Select signup button */
        const signUpButton = wrapper.find(`[data-testid='signup-button']`);

        /** Simulate click for signup button */
        signUpButton.simulate('click');

        /** Assert that fetchSignUp have been called */
        expect(fetchSignUp).toHaveBeenCalled();
    });
});
