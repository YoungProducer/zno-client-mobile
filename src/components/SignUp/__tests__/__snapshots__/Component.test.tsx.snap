// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp component Is matches snapshot 1`] = `
<div
  className="auth-background"
>
  <WithStyles(ForwardRef(Container))
    className="makeStyles-container-2"
    maxWidth="lg"
  >
    <WithStyles(ForwardRef(Paper))
      className="makeStyles-root-3"
      elevation={0}
      square={true}
    >
      <div
        className="makeStyles-logoContainer-5"
      >
        <Component />
      </div>
      <div
        className="makeStyles-innerContainer-4"
      >
        <WithStyles(ForwardRef(Grid))
          alignItems="center"
          className="makeStyles-block-9"
          container={true}
          direction="column"
          item={true}
          spacing={3}
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Component
              color="primary"
              data-testid="signup-email-input"
              error={false}
              helperText=""
              onChange={[Function]}
              placeholder="Емеїл"
              startAdornment={
                <Memo(EmailIcon)
                  className="makeStyles-icon-12"
                />
              }
              type="email"
              value=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Component
              color="primary"
              data-testid="signup-password-input"
              error={false}
              helperText=""
              onChange={[Function]}
              placeholder="Пароль"
              startAdornment={
                <Memo(LockIcon)
                  className="makeStyles-icon-12"
                />
              }
              type="password"
              value=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <Component
              color="primary"
              data-testid="signup-conf-password-input"
              error={false}
              helperText=""
              onChange={[Function]}
              placeholder="Пароль підтвердження"
              startAdornment={
                <Memo(LockIcon)
                  className="makeStyles-icon-12"
                />
              }
              type="password"
              value=""
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
          >
            <WithStyles(ForwardRef(Button))
              className="makeStyles-button-10"
              color="primary"
              data-testid="signup-button"
              disableElevation={true}
              onClick={[Function]}
              variant="contained"
            >
              Зареєструватися
            </WithStyles(ForwardRef(Button))>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))>
            <WithStyles(ForwardRef(Typography))
              align="center"
            >
              Вже зареєстровані?
              <NavigationLink
                className="makeStyles-link-11"
                navLink={
                  Object {
                    "to": "/auth/signin",
                  }
                }
              >
                Увійти
              </NavigationLink>
            </WithStyles(ForwardRef(Typography))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </div>
    </WithStyles(ForwardRef(Paper))>
  </WithStyles(ForwardRef(Container))>
</div>
`;
